packages/icons/src/components/ProductIcon/index.tsx

```diff
@@ -0,0 +1,75 @@
+import styled from '@emotion/styled'
+import { PRODUCT_ICONS } from './Icons'
+
+type Variants = 'primary' | 'danger' | 'warning'
+
+const SIZES = {
+  small: 32,
+  medium: 40,
+  large: 48,
+  xlarge: 64,
+}
+
+type ProductIconProps = {
+  name: keyof typeof PRODUCT_ICONS
+  variant?: Variants
+  disabled?: boolean
+  size?: keyof typeof SIZES
+}
+
+const StyledIcon = styled('svg', {
+  shouldForwardProp: prop => !['variant', 'disabled'].includes(prop),
+})<{ variant: Variants; disabled?: boolean; size: keyof typeof SIZES }>`
+  & {
+    width: ${({ size }) => `${SIZES[size]}px`};
+    height: ${({ size }) => `${SIZES[size]}px`};
+    min-width: ${({ size }) => `${SIZES[size]}px`};
+    max-height: ${({ size }) => `${SIZES[size]}px`};
+  }
+
+  .fill,
+  .fill > * {
```

Comment:
Are we sure abut the wildcard selector ?

Reply:
@chambo-e this has been imported from main project I haven't changed anything, the selector is correct as we use classes for filling the color depending on theme